### DISCUSSION TITLE
Line numbers become out-of-date after a few commits; How to reference functions/symbols (scroll-to-text)?

### DISCUSSION THREAD

**Discussion by SwuduSusuwu** [community/community#131838] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

https://stackoverflow.com/questions/18220650/github-link-to-function-in-source says that because it is too complex for GitHub to parse syntaxes, you must use the browser's "_scroll-to-text_", but lots of browsers do not support _scroll-to-text_: https://caniuse.com/url-scroll-to-text-fragment
On Android, _Firefox_ is stuck at the top of the page if you goto https://github.com/SwuduSusuwu/SubStack/blob/trunk/cxx/main.cxx#:~:text=testHarnesses

The problem with URLs which reference a specific _line-number_ + _commit_ is that the source file becomes out-of-date after the next commit, plus, if you do not notice that the URL followed was to a _static commit_, you end up browsing an out-of-date _snapshot_ of the repository, as opposed to the newest

**Comment by SwuduSusuwu** [comment]
Github parses most languages (to highlight syntaxes.) Asof 6/2/2024, GitHub parses C++ (presume parses most languages); if you click "_testHarnesses_", the symbols panel launches, with "testHarnesses" defined as a _function_, thus can you do _scroll-to-text_ (or _scroll-to-symbol_) with whatever parses the syntax?

**Comment by SwuduSusuwu** [comment]
Response to https://bugzilla.mozilla.org/show_bug.cgi?id=1907336 is to set `dom.text_fragments.enabled=true` in `about:config` (_Firefox_) or in `chrome://geckoview/content/config.xhtml` (_Fenix_). This fixes Firefox's `#:~:text`

**Comment by SwuduSusuwu** [comment]
This was a fix (use such as https://github.com/SwuduSusuwu/SubStack/blob/trunk/cxx/main.cxx#:~:text=testHarnesses) but now it redirects to https://github.com/SwuduSusuwu/SubStack/blob/trunk/cxx/main.cxx
What to do?

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/community/discussions/131838#discussioncomment-10401312

@SwuduSusuwu, perhaps https://github.com/microsoft/vscode/issues/216891#issue-2367970547 is of interest, especially https://github.com/orgs/community/discussions/129450#discussion-6849971.
